### DISCUSSION TITLE
why detekt need single quotation in string? (pattern & ignoreNumbers)

### DISCUSSION THREAD

**Discussion by seunggabi** [detekt/detekt#5000] [Q&A] [answered]
I know yaml string, not need quotation.
but, detekt.yaml need to quotation..

can we remove this quotation in the future?

![스크린샷 2022-06-27 오후 7 20 28](https://user-images.githubusercontent.com/8468371/175933737-8aa98b3f-f076-4576-ab39-6ba7ded85ec9.png)
![스크린샷 2022-06-27 오후 7 20 56](https://user-images.githubusercontent.com/8468371/175933748-736f3533-6173-42d1-bbab-07078c7977f0.png)
![스크린샷 2022-06-27 오후 7 20 52](https://user-images.githubusercontent.com/8468371/175933753-6faccc92-4496-44bd-a3fe-ebe8d8e8a451.png)

**Comment by BraisGabin** [comment, marked as answer]
Sure! We support the standard. The quotes are a good practice to avoid issues with special characters and that's the reason our default configuration add them.

So, if you don't like them, feel free to remove them from your configuration file. It should work.

**Comment by seunggabi** [comment]
@BraisGabin 
oh I see, Thx quick reply.
but I don't like `ignoreNumbers` even..

![스크린샷 2022-06-27 오후 7 20 56](https://user-images.githubusercontent.com/8468371/175934260-503fe68d-f49c-4dcc-a34b-948e6f1752a8.png)

Can I make this pr? `'-1'` -> `-1`

**Reply by marschwar** [reply]
I think quoting all list values was an explicit design choice. Configuration parameter values can be boolean, integer or list of strings. To be safe and not run into issues with special characters we always add the quotation marks. I do not see any benefit from changing it.

**Reply by BraisGabin** [reply]
That refactor isn't trivial on our code. And as @marschwar says it provides really little value.

**Reply by seunggabi** [reply]
@marschwar @BraisGabin
I agree thx to comment.
close this Q&A~
